code/modules/implant_hacking/board.dm

```diff
@@ -0,0 +1,288 @@
+/datum/hacking_minigame/proc/check_connections()
+
+	for(var/i in 1 to board_size)
+		for(var/k in 1 to board_size)
+			board[i][k].visited = FALSE
+			board[i][k].connected = FALSE
+
+	var/xcord
+	var/ycord
+
+	var/list/dirlist
+
+	var/list/queued_coord_list = list(list(1,1))
+
+	var/counter = 0
+
+	var/list/possible_directions = list()
+
+	while(TRUE)
+
+		if(queued_coord_list.len == counter)
+			break
+
+		if(counter > board_size*board_size)
+			stack_trace("Counter runoff stopped occured!")
+			break
+
+		possible_directions = list()
+
+		counter++
+
+		xcord = queued_coord_list[counter][1]
+		ycord = queued_coord_list[counter][2]
+
+		var/list/maybe_dir = board[xcord][ycord].get_dir_val_list()
+
+		if(xcord > 1 && ((WEST in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += WEST
+
+		if(xcord < board_size && ((EAST in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += EAST
+
+		if(ycord > 1 && ((NORTH in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += NORTH
+
+		if(ycord < board_size && ((SOUTH in maybe_dir) || board[xcord][ycord].start || board[xcord][ycord].finish))
+			possible_directions += SOUTH
+
+		board[xcord][ycord].connected = FALSE
+
+		for(var/dir in possible_directions)
+
+			switch(dir)
+
+				if(NORTH)
+
+					dirlist = board[xcord][ycord-1].get_dir_val_list()
+
+					if(board[xcord][ycord-1].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(SOUTH in dirlist)
+						if(!board[xcord][ycord-1].visited)
+							queued_coord_list += list(list(xcord,ycord-1))
+						board[xcord][ycord].connected = TRUE
+
+				if(SOUTH)
+
+					dirlist = board[xcord][ycord+1].get_dir_val_list()
+
+					if(board[xcord][ycord+1].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(NORTH in dirlist)
+						if(!board[xcord][ycord+1].visited)
+							queued_coord_list += list(list(xcord,ycord+1))
+						board[xcord][ycord].connected = TRUE
+
+				if(WEST)
+
+					dirlist = board[xcord-1][ycord].get_dir_val_list()
+
+					if(board[xcord-1][ycord].finish)
+						board[xcord][ycord].connected = TRUE
+						return TRUE
+
+					if(EAST in dirlist)
+						if(!board[xcord-1][ycord].visited)
+							queued_coord_list += list(list(xcord-1,ycord))
+						board[xcord][ycord].connected = TRUE
+
+				if(EAST)
+
+					dirlist = board[xcord+1][ycord].get_dir_val_list()
+
+					if(board[xcord+1][ycord].finish)
+						return TRUE
+
+					if(WEST in dirlist)
+						if(!board[xcord+1][ycord].visited)
+							queued_coord_list += list(list(xcord+1,ycord))
+						board[xcord][ycord].connected = TRUE
+
+		board[xcord][ycord].visited = TRUE
+
+	return FALSE
+/**
+ * Simple holder of information and some procs relating to this information
+ *
+ * Holds relevant information about a single cell of the board.
+ */
+/datum/hacking_minigame_piece
+	var/rotation = 0
```

Comment:
wouldn't it be easier if you just use the default defines for cardinals and bitshift them to go from 1 -> 2 -> 4 -> 8 and check if(WEST) rotation = NORTH

Reply:
look below